### ISSUE TITLE
FasterRCNN example: total_loss, bbox_loss and label_loss is  NaN 

### ISSUE THREAD

**Issue by Mykheievskyi** [tensorpack/tensorpack#1051]
If you're asking about an unexpected problem you met, use this template.
__PLEASE DO NOT DELETE THIS TEMPLATE, FILL IT__:

### 1. What you did:

> **I'm try run training faster rcnn on MS COCO dataset. But get losses NaN.**

(1) **If you're using examples, what's the command you run:**
```
python train.py --config \
    MODE_MASK=False \
    MODE_FPN=True \
    FPN.CASCADE=True \
    BACKBONE.RESNET_NUM_BLOCKS=[3,4,23,3] \
    TEST.RESULT_SCORE_THRESH=1e-4 \
    PREPROC.TRAIN_SHORT_EDGE_SIZE=[640,800] \
    TRAIN.LR_SCHEDULE=[420000,500000,540000] \
    DATA.BASEDIR=/media/sdd/Public_Datasets/MSCOCO \
    BACKBONE.WEIGHTS=/media/sdc/d_mykheievskyi/pretrained_models/tensorpack/ImageNet-R101-AlignPadding.npz
```

(2) **If you're using examples, have you made any changes to the examples? Paste them here:**

> **Default config, only set TRAIN.NUM_GPUS = 2**

(3) **If not using examples, tell us what you did here:**

Note that we may not be able to investigate it if there is no reproducible code.
It's always better to paste what you did instead of describing them.

### 2. What you observed:

(1) **Include the ENTIRE logs here:**

[log.log](https://github.com/tensorpack/tensorpack/files/2760594/log.log)

(2) **Other observations, if any:**
For example, CPU/GPU utilization, output images, tensorboard curves, if relevant to your issue.

### 3. What you expected, if not obvious.

If you expect higher accuracy, only in one of the two conditions can we help with it:
(1) You're unable to match the accuracy documented in tensorpack examples.
(2) It appears to be a tensorpack bug.

Otherwise, how to get high accuracy is a machine learning question and is
not our responsibility to figure out.

### 4. Your environment:
  + Python version: **3.5.4**
  + TF version: **v1.9.0-0-g25c197e023 1.9.0**
  + Tensorpack version: **0.9.0.1**
  + If you're not using tensorpack under a normal command line shell (e.g.,
    using an IDE or jupyter notebook), please retry under a normal command line shell. 
  + Hardware information, e.g. number of GPUs used: **2  gtx 1080Ti** 

About efficiency issues, PLEASE first read http://tensorpack.readthedocs.io/en/latest/tutorial/performance-tuning.html

**Comment by ppwwyyxx** [comment]
I cannot reproduce your issue when running with the same config on 2 GPUs. It's best to include your cuda version, cudnn version as well and try upgrading cudnn to a newer version. Also it may help to check that your dataset looks reasonable, your opencv is able correctly read images.

Also note that running with 2 GPUs is not how the models in the model zoo or in the paper are trained. It may require some parameter tuning and we cannot help with tuning parameters. You may try to use a smaller initial learning rate (`WARMUP_INIT_LR`), for example.

**Comment by ppwwyyxx** [comment]
It does look like you may have some issues with your dataset, or the data loading code. My run prints the following in the log:
```
| class          |   #box |                                                                                                                                                                         |:---------------|-------:|                                                                       
| BG             |      0 |                                                                                                                                                                         | person         | 257221 |                                                                       
| bicycle        |   7056 |                                                                                                                                                                         | car            |  43532 |                                                                       
| motorcycle     |   8654 |                                                                       
| airplane       |   5129 |                                                                                                                                                                         | bus            |   6061 |                                                                                                                                                                         | train          |   4570 |                                                                       
| truck          |   9970 |                                                                       
| boat           |  10575 |                                                                       
| traffic light  |  12834 |                      
| fire hydrant   |   1865 |                      
| stop sign      |   1983 |                      
| parking meter  |   1283 |                      
| bench          |   9820 |                      
| bird           |  10512 |                      
| cat            |   4766 |                      
| dog            |   5500 |                      
| horse          |   6567 |                      
| sheep          |   9223 |                      
| cow            |   8014 |                      
| elephant       |   5484 |                      
| bear           |   1294 |                      
| zebra          |   5269 |                      
| giraffe        |   5128 |                      
| backpack       |   8713 |                      
| umbrella       |  11265 |                      
| handbag        |  12340 |                      
| tie            |   6445 |
| suitcase       |   6112 |
| frisbee        |   2681 |
| skis           |   6620 |
| snowboard      |   2679 |
| sports ball    |   6291 |
| kite           |   8792 |
| baseball bat   |   3273 |
| baseball glove |   3742 |
| skateboard     |   5536 |
| surfboard      |   6093 |
| tennis racket  |   4803 |
| bottle         |  24070 |
| wine glass     |   7839 |
| cup            |  20574 |
| fork           |   5474 |
| knife          |   7759 |
| spoon          |   6159 |
| bowl           |  14323 |
```
which is quite different from what's printed in your logs. Yours have many classes with no training examples.

**Comment by ppwwyyxx** [comment]
After looking closer it appears that you probably have made changes to the code and the COCO id mapping was messed up.

**Comment by Mykheievskyi** [comment]
I solved the problem. You are rights, problem was  with  not correct matching  `COCO_id_to_category_id` and  `class_names`.

Thank you for the help.
